### ISSUE TITLE
Profiling code for low level CUDA analysis  ?

### ISSUE THREAD

**Issue by mitkotak** [maxxxzdn/jax-geometric#1]
Coming here after reading [this](https://twitter.com/maxxxzdn/status/1775237284343812312) twitter thread and the results look super cool ! 

Is it possible to share the profiling scripts for both of the plots referenced there ? I was interested in plugging it into NVIDIA's Nsight Systems and look at the low level CUDA kernels (similar to what was done [here](https://github.com/atomicarchitects/FusionFail/tree/main)).

Thanks for sharing this work ! 

Edit: Found [this](https://github.com/haifeng-jin/keras-benchmarks/issues/2) helpful thread about benchmarking `torch.compile` which might be relevant.

**Comment by maxxxzdn** [comment]
Hi Mit, 

Sure, I will push the code soon, likely on Friday/Saturday. 
Thanks for sharing!

**Comment by mitkotak** [comment]
Apologies for reopening this but was wondering whether you have timing/benchmarking scripts that I can quickly throw onto my RTX A5500 to see what the numbers look like and hopefully do an apples to apples comparison with your plots. Thanks again for doing this !

**Comment by mitkotak** [comment]
Also was trying to get `torch.compile` to work with `fullgraph=True` and as you said on Twitter there seem to be some errors. Is it Ok if I file issues for them so that other folks can help out with debugging ?

Would be cool to compare`jax.jit` vs `torch.compile` for these type of implementations (Don't worry I am not trying to start another Twitter war :D )

**Comment by mitkotak** [comment]
And FYI for SEGNN and EGNN there's also [these](https://github.com/smsharma/eqnn-jax/blob/main/models/segnn.py) implementations that would be interesting to compare against.

**Comment by maxxxzdn** [comment]
I just added the script for CEGNN in JAX. 

Concerning opening issues, please go ahead, I would love people to help. I was recently playing with CEGNNs and managed to indicate operations that make torch struggle with compilation:

- normalization (due to the for loop in algebra.norms, perhaps)
- broadcasting along subspaces in MVLinear
   - weight = self.weight.repeat_interleave(self.algebra.subspaces, dim=-1)
- get_weight in Cayley table

I will update the codebase if I manage to resolve them :)
